src/engine/renderer/gl_shader.cpp

```diff
@@ -22,8 +22,20 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 // gl_shader.cpp -- GLSL shader handling
 
 #include "gl_shader.h"
+#include <type_traits>
+
+// We currently write GLShaderHeader to a file and memcpy all over it.
+// Make sure it's a pod, so we don't put a std::string in it or something
+// and try to memcpy over that or binary write an std::string to a file.
+static_assert(std::is_pod<GLShaderHeader>::value, "Value must be a pod while code in this cpp file reads and writes this object to file as binary.");
```

Comment:
Good static assert, I did not know it was supported by gcc 4.6 but as travis is green it is ok. Although you migt want to replace Value in the string by GLShaderHeader